### PR TITLE
Add alternative for how to install if a change in nightly broke the rustup component

### PR DESCRIPTION
**Pull request by timotree3**
The situation described where Miri is broken on nightly [is currently the case](https://rust-lang-nursery.github.io/rust-toolstate) and I ran into this issue just now.

I'm not sure if I put this in the right section, at first I considered just putting it right after the command `rustup component add miri`.

[Prior art](https://github.com/rust-lang/rls/blob/57347657edf6f55bf75423c4d34094543d165b81/README.md#error-component-rls-is-unavailable-for-download-nightly)

### CODE REVIEW

**Comment by timotree3** [comment]
Another option would be to instead suggest using an earlier nightly and link to [the component history](https://rust-lang.github.io/rustup-components-history/).

**Comment by RalfJung** [comment]
Thanks for the PR! :)

However, I think I'd prefer to keep this consistent between tools and instead tell people to use an old nightly.  Eventually, rustup will have better support for this as well.

This repo will not always work with the latest nightly either, fixes first have to land here. Also, we have plans to move this repo to test against rustc master instead of a nightly, so even the latest nightly might not be "new enough" for Miri.

What do you think, @oli-obk?

**Comment by oli-obk** [comment]
Jop, I agree, we should not try to work around this. We'll find a convenient solution on the rustup side for nightly users.

**Comment by RalfJung** [comment]
No need to close; @timotree3 could you implement what you suggested?

> Another option would be to instead suggest using an earlier nightly and link to the component history.

**Comment by timotree3** [comment]
@RalfJung how's that?

**Comment by RalfJung** [comment]
That's a lot of text in the middle of the quick-start tutorial, and also now it looks like the part starting "Now you can run your project in Miri:" belongs to the "error" section.

I'd suggest to just add a short paragraph instead of this entire section, saying something like

> If `rustup` says the `miri` component is unavailable, that's because not all nightly releases come with all tools. Check out [this website](https://rust-lang.github.io/rustup-components-history/) to determine a nightly version that comes with Miri and install that, e.g. using `rustup install nightly-2018-12-06`.

File: README.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ Install Miri via `rustup`:
 rustup component add miri
 ```
 
+If `rustup` says the `miri` component is unavailable, that's because not all nightly releases come with all tools. Check out this website to determine a nightly version that comes with Miri and install that, e.g. using `rustup install nightly-2019-03-28`.
```
Comment (RalfJung) [inline README.md]: "this website" should link to the tool state history.

**Comment by timotree3** [comment]
So will this get merged then?

**Comment by RalfJung** [comment]
Yes, it will... thanks for the reminder. Unfortunately we don't have bors here so someone will have to remember to click the button after CI is green.^^

**Pull request merged**